### PR TITLE
Add HermitianMatrixSpace

### PR DESCRIPTION
**Pull request by blegat**
Closes https://github.com/jump-dev/JuMP.jl/issues/3291

- [x] Add example in manual like skew symmetric
- [x] Tests (including edge cases with variable bounds, etc...)

### CODE REVIEW

File: test/JuMPExtension.jl [hunk 1]
```diff
@@ -426,6 +434,20 @@ function JuMP.is_valid(model::MyModel, constraint_ref::MyConstraintRef)
...
+    ::Type{F},
+    ::Type{S},
+) where {F,S<:JuMP.MOI.AbstractSet}
+    constraints = JuMP.ConstraintRef[]
+    for (index, c) in model.constraints
+        if JuMP.jump_function(c) isa F && JuMP.moi_set(c) isa S
+            push!(constraints, JuMP.ConstraintRef(model, index, JuMP.shape(c)))
+        end
+    end
+    return connstraints
```
Comment (odow) [inline test/JuMPExtension.jl:448]:
```suggestion
    return constraints
```

**Review by odow — APPROVED** [review]
